File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/MatchedBet.java
package com.umbrella.insurance.core.models.entities;

import com.umbrella.insurance.core.deserializers.TimestampDeserializer;
import com.umbrella.insurance.core.serializers.TimestampSerializer;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.sql.Timestamp;

@Getter
@Setter
@Entity
@Table(name = "matched_bets", schema = "public")
public class MatchedBet {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "matched_bet_id", nullable = false)
    private Long id;

    @OneToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "pending_bet_id_1")
    private PendingBet pendingBet1;

    @OneToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "pending_bet_id_2")
    private PendingBet pendingBet2;

    @NotNull
    @Column(name = "odds_1", nullable = false)
    private Double odds1;

    @NotNull
    @Column(name = "odds_2", nullable = false)
    private Double odds2;

    @NotNull
    @Column(name = "wager_amount_1", nullable = false)
    private Double wagerAmount1;

    @NotNull
    @Column(name = "wager_amount_2", nullable = false)
    private Double wagerAmount2;

    @JsonSerialize(using = TimestampSerializer.class)
    @JsonDeserialize(using = TimestampDeserializer.class)
    @NotNull
    @Column(name = "created_date_time", nullable = false)
    private Timestamp createdDateTime;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "fee_id_1")
    private Fee feeId1;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "fee_id_2")
    private Fee feeId2;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "matched_bet_state_id")
    private MatchedBetState matchedBetState;

}

File: umbrella-insurance-react-ui/src/models/users/bets/matchedBets/v1/MatchedBet.ts
import { Fee } from "../../../fees/v1/Fee";
import { PendingBet } from "../../pendingBets/v1/PendingBet";
import { MatchedBetState } from "../matchedBetStates/v1/MatchedBetState";

export class MatchedBet {
    id?: number;
    createdDateTime?: string;
    pendingBet1?: PendingBet;
    pendingBet2?: PendingBet;
    odds1?: number;
    odds2?: number;
    wagerAmount1?: number;
    wagerAmount2?: number;
    fee1?: Fee;
    fee2?: Fee;
    matchedBetState?: MatchedBetState;

    constructor(obj?:any) {
        if(obj) {
            if(obj?.id) {
                this.id = obj.id;
            }
            if(obj?.createdDateTime) {
                this.createdDateTime = obj.createdDateTime;
            }
            if(obj?.pendingBet1) {
                this.pendingBet1 = new PendingBet(obj.pendingBet1);
            }
            if(obj?.pendingBet2) {
                this.pendingBet2 = new PendingBet(obj.pendingBet2);
            }
            if(obj?.odds1) {
                this.odds1 = obj.odds1;
            }
            if(obj?.odds2) {
                this.odds2 = obj.odds2;
            }
            if(obj?.wagerAmount1) {
                this.wagerAmount1 = obj.wagerAmount1;
            }
            if(obj?.wagerAmount2) {
                this.wagerAmount2 = obj.wagerAmount2;
            }
            if(obj?.fee1) {
                this.fee1 = new Fee(obj.fee1);
            }
            if(obj?.fee2) {
                this.fee2 = new Fee(obj.fee2);
            }
            if(obj?.matchedBetState) {
                this.matchedBetState = new MatchedBetState(obj.matchedBetState);
            }
        }
    }
}